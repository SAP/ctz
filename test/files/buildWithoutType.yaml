repository: docker.io/username
tag: latest
before-all:
  - npx cds build --production
modules:
  - name: bookshop-approuter
    tag: v1
    build-parameters:
      buildpack:
        builder: builder-jammy-base
        path: app
        env:
          BP_NODE_RUN_SCRIPTS: ""
  - name: dockerfile-image
    build-parameters:
      dockerfile: dockerfiles/sample.Dockerfile
  - name: bookshop-hana-deployer
    build-parameters:
      commands:
        - 'pack build bookshop-hana-deployer --path db --builder paketobuildpacks/builder-jammy-base'