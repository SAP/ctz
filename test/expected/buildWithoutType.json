{
    "commands": [
        {
            "name": "bookshop-approuter",
            "tag": "v1",
            "buildCmd": [
                [
                    "pack",
                    "build",
                    "bookshop-approuter:v1",
                    "--path",
                    "app",
                    "--builder",
                    "paketobuildpacks/builder-jammy-base",
                    "--env",
                    "BP_NODE_RUN_SCRIPTS=\"\""
                ]
            ],
            "tagCmd": [
                "docker",
                "tag",
                "bookshop-approuter:v1",
                "docker.io/username/bookshop-approuter:v1"
            ],
            "pushCmd": [
                "docker",
                "push",
                "docker.io/username/bookshop-approuter:v1"
            ]
        },
        {
            "name": "dockerfile-image",
            "tag": "latest",
            "buildCmd": [
                [
                    "docker",
                    "build",
                    "-t",
                    "dockerfile-image:latest",
                    "-f",
                    "dockerfiles/sample.Dockerfile",
                    "."
                ]
            ],
            "tagCmd": [
                "docker",
                "tag",
                "dockerfile-image:latest",
                "docker.io/username/dockerfile-image:latest"
            ],
            "pushCmd": [
                "docker",
                "push",
                "docker.io/username/dockerfile-image:latest"
            ]
        },
        {
            "name": "bookshop-hana-deployer",
            "tag": "latest",
            "buildCmd": [
                [
                    "pack",
                    "build",
                    "bookshop-hana-deployer",
                    "--path",
                    "db",
                    "--builder",
                    "paketobuildpacks/builder-jammy-base"
                ]
            ],
            "tagCmd": [
                "docker",
                "tag",
                "bookshop-hana-deployer:latest",
                "docker.io/username/bookshop-hana-deployer:latest"
            ],
            "pushCmd": [
                "docker",
                "push",
                "docker.io/username/bookshop-hana-deployer:latest"
            ]
        }
    ],
    "before_all": [
        [
            "npx",
            "cds",
            "build",
            "--production"
        ]
    ],
    "repository": "docker.io/username"
}